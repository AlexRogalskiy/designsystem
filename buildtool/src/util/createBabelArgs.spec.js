const createBabelArgs = require('./createBabelArgs');

describe('createBabelArgs', () => {
    test('minimal args', () =>
        expect(
            createBabelArgs({
                source: 'sourceDir',
                cjs: 'cjsDir',
                es: 'esDir',
            }),
        ).toMatchSnapshot());

    test('cjs only', () =>
        expect(
            createBabelArgs({
                source: 'sourceDir',
                cjs: 'cjsDir',
                es: false,
            }),
        ).toMatchSnapshot());

    test('esm only', () =>
        expect(
            createBabelArgs({
                source: 'sourceDir',
                cjs: false,
                es: 'esDir',
            }),
        ).toMatchSnapshot());
});
