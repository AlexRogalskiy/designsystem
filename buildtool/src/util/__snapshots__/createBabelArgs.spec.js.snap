// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createBabelArgs cjs only 1`] = `
Array [
  Array [
    "babel",
    "--config-file",
    "config/babel.config.js",
    "--ignore",
    "**/*.spec.js",
    "--env-name",
    "cjs",
    "-d",
    "cjsDir",
    "sourceDir",
  ],
]
`;

exports[`createBabelArgs esm only 1`] = `
Array [
  Array [
    "babel",
    "--config-file",
    "config/babel.config.js",
    "--ignore",
    "**/*.spec.js",
    "--env-name",
    "es",
    "-d",
    "esDir",
    "sourceDir",
  ],
]
`;

exports[`createBabelArgs minimal args 1`] = `
Array [
  Array [
    "babel",
    "--config-file",
    "config/babel.config.js",
    "--ignore",
    "**/*.spec.js",
    "--env-name",
    "es",
    "-d",
    "esDir",
    "sourceDir",
  ],
  Array [
    "babel",
    "--config-file",
    "config/babel.config.js",
    "--ignore",
    "**/*.spec.js",
    "--env-name",
    "cjs",
    "-d",
    "cjsDir",
    "sourceDir",
  ],
]
`;
